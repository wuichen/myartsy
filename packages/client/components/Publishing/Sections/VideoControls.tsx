import React from "react"
import styled, { StyledFunction } from "styled-components"

interface VideoControlsProps extends React.HTMLProps<HTMLDivElement> {
  mini?: boolean
}

export const VideoControls: React.SFC<VideoControlsProps> = props => {
  const { mini } = props

  return (
    <PlayButton mini={mini} className="PlayButton">
      <PlayButtonCaret className="PlayButton__PlayButtonCaret" mini={mini} />
    </PlayButton>
  )
}

const Div: StyledFunction<
  VideoControlsProps & React.HTMLProps<HTMLDivElement>
> = styled.div

const PlayButtonCaret = Div`
  color: black;
  border-top: ${props => (props.mini ? "13" : "20")}px solid transparent;
  border-bottom: ${props => (props.mini ? "13" : "20")}px solid transparent;
  border-left: ${props => (props.mini ? "20" : "30")}px solid black;
`

const PlayButton = Div`
  background: white;
  width: ${props => (props.mini ? "50px" : "70px")};
  height: ${props => (props.mini ? "50px" : "70px")};
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 2;
  cursor: pointer;
  border: 0;
  outline: 0;
`
