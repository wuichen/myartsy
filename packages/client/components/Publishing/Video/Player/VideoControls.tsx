import { garamond } from "Assets/Fonts"
import React, { Component } from "react"
import styled from "styled-components"
import { IconVideoFullscreen } from "../../Icon/IconVideoFullscreen"
import { MuteUnmute, MuteUnmuteContainer } from "./MuteUnmute"
import { PlayPause } from "./PlayPause"
import { Scrubber } from "./Scrubber"
import { Timestamp } from "./Timestamp"

interface Props extends React.HTMLProps<HTMLDivElement> {
  isMuted: boolean
  isPlaying: boolean
  title: string
  duration: number
  currentTime: number
  toggleFullscreen: () => void
  toggleMute: () => void
  togglePlay: () => void
  pause: () => void
  play: () => void
  seekTo: (v) => void
}

export class VideoControls extends Component<Props, null> {
  render() {
    const {
      isMuted,
      isPlaying,
      title,
      duration,
      currentTime,
      toggleFullscreen,
      toggleMute,
      togglePlay,
      pause,
      play,
      seekTo,
    } = this.props

    return (
      <VideoControlsContainer>
        <TopControls>
          <Block>
            <PlayPause togglePlay={togglePlay} isPlaying={isPlaying} />
            <Title>{title}</Title>
          </Block>
          <Block>
            <Timestamp currentTime={currentTime} duration={duration} />
            <MuteUnmute toggleMute={toggleMute} isMuted={isMuted} />
            <IconVideoFullscreen onClick={toggleFullscreen} color="white" />
          </Block>
        </TopControls>
        <Scrubber
          duration={duration}
          currentTime={currentTime}
          pause={pause}
          play={play}
          seekTo={seekTo}
          isPlaying={isPlaying}
        />
      </VideoControlsContainer>
    )
  }
}

const TopControls = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`
export const VideoControlsContainer = styled.div`
  color: white;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid white;
  border-radius: 2px;
  padding: 20px;
  margin: 20px;
  opacity: 0;
  transition: opacity 0.25s ease;

  ${Timestamp} {
    margin-right: 30px;
  }

  ${MuteUnmuteContainer} {
    margin-right: 20px;
  }

  ${IconVideoFullscreen} {
    cursor: pointer;
  }
`
const Title = styled.div`
  ${garamond("s23")};
  margin-left: 20px;
`
const Block = styled.div`
  display: flex;
`
