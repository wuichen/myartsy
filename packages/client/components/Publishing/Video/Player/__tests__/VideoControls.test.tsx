import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { VideoControls } from "../VideoControls"

describe("VideoControls", () => {
  it("renders properly", () => {
    const videoControls = renderer
      .create(
        <VideoControls
          isMuted={false}
          isPlaying={false}
          title="Video Title"
          duration={5000}
          currentTime={0}
          toggleFullscreen={jest.fn()}
          toggleMute={jest.fn()}
          togglePlay={jest.fn()}
          pause={jest.fn()}
          play={jest.fn()}
          seekTo={jest.fn()}
        />
      )
      .toJSON()
    expect(videoControls).toMatchSnapshot()
  })
})
