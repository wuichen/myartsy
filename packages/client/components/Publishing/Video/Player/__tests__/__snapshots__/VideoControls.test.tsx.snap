// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoControls renders properly 1`] = `
.c11 {
  width: 28px;
  height: 28px;
}

.c9 {
  height: 28px;
  cursor: pointer;
}

.c4 {
  width: 32px;
  height: 32px;
}

.c3 {
  height: 28px;
  width: 28px;
  cursor: pointer;
}

.c12 {
  width: 100%;
  background-size: 0% 100%;
  height: 18px;
  margin-top: 15px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
}

.c12::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c12::-ms-track {
  width: 100%;
  height: 2px;
  cursor: pointer;
  background: transparent;
  border: 0px;
  color: transparent;
}

.c12::-webkit-slider-runnable-track {
  height: 2px;
  cursor: pointer;
  -webkit-animate: 0.2s;
  animate: 0.2s;
  background-color: white;
  border: 0px;
}

.c12::-moz-range-track {
  height: 2px;
  cursor: pointer;
  -webkit-animate: 0.2s;
  animate: 0.2s;
  background-color: white;
  border: 0px;
}

.c12::-webkit-slider-thumb {
  color: white;
  border: 0px;
  height: 12px;
  width: 12px;
  border-radius: 50px;
  background: white;
  cursor: pointer;
  margin-top: -5px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c12::-moz-range-thumb {
  color: white;
  box-shadow: 0px 0px 0px white;
  border: 0px;
  height: 12px;
  width: 12px;
  border-radius: 50px;
  background: white;
  cursor: pointer;
  margin-top: -5px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c12::-ms-fill-lower {
  background: white;
  border: 0px;
}

.c12::-ms-fill-upper {
  background: white;
  border: 0px;
}

.c12::-ms-thumb {
  color: white;
  box-shadow: 0px 0px 0px white;
  border: 0px;
  height: 12px;
  width: 12px;
  border-radius: 50px;
  background: white;
  cursor: pointer;
  margin-top: -5px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c12:focus::-webkit-slider-runnable-track {
  background: white;
}

.c12:focus {
  outline: none;
}

.c7 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

.c0 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid white;
  border-radius: 2px;
  padding: 20px;
  margin: 20px;
  opacity: 0;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c0 .c6 {
  margin-right: 30px;
}

.c0 .c8 {
  margin-right: 20px;
}

.c0 .c10 {
  cursor: pointer;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  margin-left: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        onClick={[MockFunction]}
      >
        <svg
          className="c4"
          version="1.1"
          viewBox="0 0 40 56"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <polygon
              fill="white"
              points="0 0 0 56 39.3600006 28"
            />
          </g>
        </svg>
      </div>
      <div
        className="c5"
      >
        Video Title
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c6 c7"
      >
        00:00
         / 
        23:20
      </div>
      <div
        className="c8 c9"
        onClick={[MockFunction]}
      >
        <svg
          height="30px"
          width="30px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="white"
          >
            <path
              d="M19.1285714,15.3428571 L19.1285714,28.7571429 L11.8285714,21.4714286 L0.757142857,21.4714286 L0.757142857,8.04285714 L11.8285714,8.04285714 L19.1285714,0.757142857 L19.1285714,15.3428571 Z"
            />
          </g>
        </svg>
      </div>
      <svg
        className="c10 c11"
        height="32px"
        onClick={[MockFunction]}
        viewBox="0 0 28 28"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(-1151 -22)"
        >
          <rect
            height="92"
            rx="4"
            stroke="white"
            width="1200"
          />
          <path
            d="M1150 21h30v30h-30z"
          />
          <g
            fill="#FFF"
          >
            <path
              d="M1151 40h2v10h-2z"
            />
            <path
              d="M1161 48v2h-10v-2zM1161 22v2h-10v-2z"
            />
            <path
              d="M1153 32h-2V22h2z"
            />
            <g>
              <path
                d="M1179 32h-2V22h2z"
              />
              <path
                d="M1169 24v-2h10v2z"
              />
            </g>
            <g>
              <path
                d="M1169 50v-2h10v2z"
              />
              <path
                d="M1177 40h2v10h-2z"
              />
            </g>
          </g>
          <path
            d="M1157 28h16v16h-16z"
            fill="#FFF"
          />
        </g>
      </svg>
    </div>
  </div>
  <input
    className="c12"
    max="5000.0000"
    onChange={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    step="any"
    type="range"
    value={0}
  />
</div>
`;
